packages/massa-web3/src/experimental/client/publicAPI.ts

```diff
@@ -78,11 +85,48 @@ export class PublicAPI {
   }
 
   async executeReadOnlyCall(
-    readOnlyCall: ReadOnlyCall
-  ): Promise<ExecuteReadOnlyResponse> {
-    return this.connector
-      .execute_read_only_call([readOnlyCall])
-      .then((r) => r[FIRST])
+    readOnlyCall: ReadOnlyCallParams
+  ): Promise<ReadOnlyCallResult> {
+    const [readOnlyRes] = await this.connector.execute_read_only_call([
```

Comment:
Separation of concerns: the level of abstraction chosen for your variable name is incorrect. Using `readOnlyRes` in a *readOnlyCall* is overcomplicated. `res` or `response` should be sufficient.